nomenclature/processor/region.py

```diff
@@ -459,3 +491,40 @@ def _merge_with_provided_data(
     # merge aggregated data onto original common-region data
     index = aggregate_df._data.index.difference(common_region_df._data.index)
     return common_region_df.append(aggregate_df._data[index])
+
+
+def _check_exclude_region_overlap(values: Dict, region_type: str) -> Dict:
+    if values.get("exclude_regions") is None or values.get(region_type) is None:
+        return values
+    if overlap := set(values["exclude_regions"]) & {
+        r.name for r in values[region_type]
+    }:
+        raise ExcludeRegionOverlapError(
+            region=overlap, region_type=region_type, file=values["file"]
+        )
+    return values
+
+
+def _check_unexpected_regions(
+    df: IamDataFrame, mapping: RegionAggregationMapping
+) -> None:
+    # Raise value error if a region in the input data is not mentioned in the model
+    # mapping
+
+    if regions_not_found := set(df.region) - set(
+        mapping.model_native_region_names
+        + mapping.common_region_names
+        + [
+            const_reg
+            for comm_reg in mapping.common_regions or []
+            for const_reg in comm_reg.constituent_regions
+        ]
+        + (mapping.exclude_regions or [])
+    ):
+        raise ValueError(
+            f"Did not find region(s) {regions_not_found} in 'native_regions', "
+            "'common_regions' or 'exclude_regions' in model mapping for "
+            f"{mapping.model} in {mapping.file}. If they are not meant for upload add "
```

Comment:
I would not mention the “upload” because the package per se is independent from the IIASA Explorer infrastructure

Reply:
Ah right, good point I was thinking about it too much in terms of the Scenario Explorer.

Reply:
Updated it